File: ModalOverlayIOS-ObjC/ModalOverlayIOS-ObjC/OverlayViewController.m
//
//  OverlayViewController.m
//  Modal Test
//
//  Created by Rob Bajorek on 8/8/14.
//  Copyright (c) 2014 Rob Bajorek. All rights reserved.
//

#import "OverlayViewController.h"

@interface OverlayViewController ()
@property (weak) IBOutlet UIButton *dismissButton;
- (IBAction)optionOneTapped:(id)sender;
- (IBAction)optionTwoTapped:(id)sender;
@end

@implementation OverlayViewController

- (IBAction)optionOneTapped:(id)sender {
    [self.delegate optionOneChosen];
}

- (IBAction)optionTwoTapped:(id)sender {
    [self.delegate optionTwoChosen];
}

@end

File: ModalOverlayIOS-Swift/ModalOverlayIOS-Swift/OverlayViewController.swift
//
//  OverlayViewController.swift
//  ModalOverlayIOS-Swift
//
//  Created by Rob Bajorek on 12/22/18.
//  Copyright © 2018 Rob Bajorek. All rights reserved.
//

import UIKit

protocol OverlayViewControllerDelegate: class {
    func optionOneChosen()
    func optionTwoChosen()
}

class OverlayViewController: UIViewController {
    @IBOutlet private weak var dismissButton: UIButton!
    weak var delegate: OverlayViewControllerDelegate?

    @IBAction private func optionOneTapped(_ sender: Any) {
     delegate?.optionOneChosen()
    }

    @IBAction private func optionTwoTapped(_ sender: Any) {
        delegate?.optionTwoChosen()
    }
}
